#!/usr/bin/env python
PACKAGE = "cloud_factory"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#name, paramtype, level, description, default, min, max
#gen.add("mode", int_t, 0, "Modes available 0 = thorus, 1 = rectangle, 2 = loadFile, 3 = crazyShape", 3, 0, 3)
#gen.add("raysIncl", int_t, 0, "vertical inclination rays (layers) = cloud height", 16, 0, 50)
gen.add("inclMin", double_t, 0, "inclMin: lowest inclination angle", -0.26180, -0.785398, 0.785398)
gen.add("inclMax", double_t, 0, "inclMax: highest inclination angle", 0.261799, 0, 0.785398)
gen.add("inclRes", double_t, 0, "inclRes: resolution of inclination angle", 0.034907, 0.0, 0.1)
gen.add("azimMin", double_t, 0, "azimMin: (smallest) azimuth start angle", 0.0, 0.0, 0.00698132)
gen.add("azimMax", double_t, 0, "azimMax: (biggest) azimuth end angle", 6.28319, 0.0, 12.6)
gen.add("azimRes", double_t, 0, "azimRes: resolution of azimuth angle", 0.00349066, 0.00174533, 0.0628319)
gen.add("absBase", double_t, 0, "absBase: first radius value", 0.1, 0.1, 1.0)
gen.add("absDiff", double_t, 0, "absDiff: radius increment per loop", 0.002, 0.0, 0.5)

exit(gen.generate(PACKAGE, "cloud_factory", "theAzimuth"))